app/Platform/Commands/SyncGameSets.php

```diff
@@ -4,35 +4,91 @@
+use App\Models\GameSetLink;
+use App\Models\System;
+use App\Platform\Enums\GameSetType;
 use Illuminate\Console\Command;
 
 class SyncGameSets extends Command
 {
-    use SyncTrait;
-
     protected $signature = 'ra:sync:game-sets {--f|full} {--p|no-post}';
-    protected $description = 'Sync game alternatives to sets';
+    protected $description = 'Sync hubs to game_sets';
 
     public function __construct()
     {
         parent::__construct();
     }
 
-    /**
-     * @throws Exception
-     */
     public function handle(): void
     {
-        $this->sync('game_sets');
-    }
+        $this->info('Running a full GameAlternatives -> game_sets sync...');
+
+        // This will be a full reset. Delete any existing game_sets data.
+        $this->info("\nDeleting any existing game_sets data...");
+        GameSetLink::query()->forceDelete();
+        GameSetGame::query()->forceDelete();
+        GameSet::query()->forceDelete();
+        $this->info("Deleted all existing game_sets data.");
+
+        // Loop through all GameAlternatives and create game_sets.
+        $distinctGameIds = GameAlternative::select('gameID')->distinct()->pluck('gameID');
+        $distinctGameIdsCount = $distinctGameIds->count();
+
+        $this->info("\nUpserting {$distinctGameIdsCount} game_sets derived from legacy GameAlternatives.");
+        $progressBar = $this->output->createProgressBar($distinctGameIdsCount);
+        foreach ($distinctGameIds as $gameId) {
+            $game = Game::find($gameId);
+
+            if ($game->ConsoleID === System::Hubs) {
+                GameSet::updateOrCreate(
+                    ['game_id' => $game->id],
+                    ['title' => $game->title, 'type' => GameSetType::HUB],
+                );
+            } else {
+                GameSet::updateOrCreate(
+                    ['game_id' => $game->id],
+                    ['type' => GameSetType::GAME]
+                );
+            }
 
-    // protected function preProcessEntity(object $origin, array $transformed): array
-    // {
-    // }
+            $progressBar->advance();
+        }
+        $progressBar->finish();
 
-    // protected function postProcessEntity(object $origin, object $transformed): void
-    // {
-    // }
+        $gameAltsCount = GameAlternative::count();
+        $this->info("\nPopulating {$gameAltsCount} game_set_games and game_set_links...");
+
+        $progressBar = $this->output->createProgressBar($gameAltsCount);
+
+        foreach (GameAlternative::cursor() as $gameAlt) {
+            $parentGameSet = GameSet::firstWhere('game_id', $gameAlt->gameID);
+            $game = Game::find($gameAlt->gameID);
+            $altGame = Game::find($gameAlt->gameIDAlt);
+
+            // Only process relationships in one direction so we avoid duplicate game_set_games and game_set_links.
+            if ($gameAlt->gameID < $gameAlt->gameIDAlt) {
+                if ($game->ConsoleID === System::Hubs || $altGame->ConsoleID === System::Hubs) {
```

Comment:
I think this should be an AND. As an OR, game game_sets are being linked to the hubs:
```
select type, game_id from game_sets where id in (select parent_game_set_id from game_set_links where child_game_set_id=(select id from game_sets where game_id=6708));
```

Might be some logical reordering necessary to ensure the hub is always the parent too. See my comment in `UpdateGameSetFromGameAlternativesModification`.

Reply:
This makes sense. Latest now uses logical AND instead of OR.